components/datetime/src/pattern/reference/pattern.rs

```diff
@@ -234,3 +237,176 @@ impl Serialize for Pattern {
         }
     }
 }
+
+#[cfg(feature = "provider_serde")]
+mod week_field_serialization {
+    use crate::fields::Week;
+    use alloc::format;
+    use core::convert::TryInto;
+    use serde::{self, de, Deserialize};
+
+    pub fn serialize<S>(week: &Week, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_char((*week).into())
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Week, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        char::deserialize(deserializer)?.try_into().map_err(|err| {
+            de::Error::invalid_value(
+                de::Unexpected::Other(&format!("{}", err)),
+                &"a valid UTS 35 week symbol",
+            )
+        })
+    }
+}
+
+/// A collection of plural variants of a pattern.
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct PluralPattern {
+    /// The field that 'variants' are predicated on.
+    #[cfg_attr(feature = "provider_serde", serde(with = "week_field_serialization"))]
+    pivot_field: Week,
+    #[cfg_attr(feature = "provider_serde", serde(with = "tuple_vec_map"))]
+    variants: Vec<(PluralCategory, Pattern)>,
```

Comment:
issue: Perhaps we should use LiteMap here? That way we don't need to pull in another crate.

Eventually we should use ZeroMap here but that might be better done outside this PR since Pattern isn't ready yet.